### PR TITLE
Feature/analysis journey

### PR DESCRIPTION
**Pull request by reisingerf**
This includes some breaking changes, like the change from internal auto-id fields to OrcaBus IDs as PKs. 

There are also updates to the API with removed/added endpoints and updated/extended response schemas.

### CODE REVIEW

**Review by raylrui — APPROVED** [review]
LGTM

File: lib/workload/stateless/stacks/workflow-manager/workflow_manager/models/state.py [hunk 1]
```diff
@@ -89,27 +90,26 @@ class State(OrcaBusBaseModel):
...
 
     def __str__(self):
-        return f"ID: {self.id}, status: {self.status}"
+        return f"ID: {self.orcabus_id}, status: {self.status}"
 
     def to_dict(self):
         return {
-            "id": self.id,
-            "workflow_run_id": self.workflow_run.id,
+            "orcabusId": self.orcabus_id,
```
Comment (williamputraintan) [inline lib/workload/stateless/stacks/workflow-manager/workflow_manager/models/state.py:111]: Should this be `snake_case`?

**Comment by victorskl** [comment]
.. reviewing

File: lib/workload/stateless/stacks/workflow-manager/workflow_manager/viewsets/base.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        Since we have custom orcabus_id prefix for each model, we need to remove the prefix before retrieving it.
+        """
+        pk = self.kwargs.get('pk')
+        print("DEBUG: pk")
+        print(pk)
+        if pk and pk.startswith(self.orcabus_id_prefix):
+            pk = pk[len(self.orcabus_id_prefix):]
+
+        print("DEBUG: self.queryset")
+        print(self.queryset)
```
Comment (williamputraintan) [inline lib/workload/stateless/stacks/workflow-manager/workflow_manager/viewsets/base.py:28]: remove print? or is it intended?

**Review by williamputraintan — APPROVED** [review]
LGTM!

File: lib/workload/stateless/stacks/workflow-manager/workflow_manager/models/state.py (hunk 1 above)
Comment (reisingerf) [reply lib/workload/stateless/stacks/workflow-manager/workflow_manager/models/state.py]: Fair enough. Not even sure those methods are needed at all.
I'll just remove them.

File: lib/workload/stateless/stacks/workflow-manager/workflow_manager/viewsets/base.py (hunk 2 above)
Comment (reisingerf) [reply lib/workload/stateless/stacks/workflow-manager/workflow_manager/viewsets/base.py]: Good point!
I cleaned up the debug/logging statements a bit.

**Review by victorskl — APPROVED** [review]

**Pull request merged**